daemon/config_experimental.go

```diff
@@ -1,9 +0,0 @@
-// +build experimental
-
-package daemon
-
-import flag "github.com/docker/docker/pkg/mflag"
-
-func (config *Config) attachExperimentalFlags(cmd *flag.FlagSet, usageFn func(string) string) {
-	cmd.StringVar(&config.DefaultNetwork, []string{"-default-network"}, "", usageFn("Set default network"))
```

Comment:
--default-network being a daemon option, and it is not synchronized across the cluster, it is quite easy to have different --default-networks in different nodes in the cluster. Hence, it is upto the user to make sure it is consistent. That results in unexpected behaviour when containers are launched in different nodes.
Also, what is the expectation if the default-network changes after a daemon restart ?
Should the stopped containers start using the new network as the default-network or stick to the older one ?

Now that swarm and compose will have native integration with the new networking features, it is appropriate for swarm to make sure it creates the networks it want to use before using it.

Thats the reason for the removal.